nimare/transforms.py

```diff
@@ -187,6 +188,67 @@ def resolve_transforms(target, available_data, masker):
         return None
 
 
+def images_to_coordinates(
+    images_df,
+    z_threshold,
+    space,
+    masker,
+    cluster_threshold=None,
+    min_distance=8.0,
+):
+    """Extract peak statistical coordinates from statistical z or p maps
+
+    Parameters
+    ----------
+    images_df : :class:`pandas.DataFrame`
+        DataFrame with paths to images for studies in Dataset.
+    z_threshold : :obj:`float`
+        Cluster forming z-scale threshold in `stat_img`.
+    space : :obj:`string`
+        the final frontier
+    masker : :class:`nilearn.input_data.NiftiMasker`
+        masker for dataset
+    cluster_threshold : :obj:`int` or `None`, optional
+        Cluster size threshold, in voxels.
+    min_distance : :obj:`float`, optional
+        Minimum distance between subpeaks in mm. Default=8mm.
+
+    Returns
+    -------
+    coordinates_df : :class:`pandas.DataFrame`
+        DataFrame containing statistical peaks.
+    """
+
+    coordinates_dict = {}
+    for _, row in images_df.iterrows():
+        if row.get("z"):
+            clusters = get_clusters_table(
+                nib.load(row.get("z")), z_threshold, cluster_threshold, min_distance
+            )
+        elif row.get("p"):
+            LGR.warning(f"No Z map for {row['id']}, using p map")
+            p_threshold = z_to_p(z_threshold)
+            clusters = get_clusters_table(
+                nib.load(row.get("p")), p_threshold, cluster_threshold, min_distance
+            )
+        else:
+            raise ValueError(f"{row['id']} needs either a Z map or p map")
+        coordinates_dict[row["study_id"]] = {
+            "contrasts": {
+                row["contrast_id"]: {
+                    "coords": {
+                        "space": space,
+                        "x": list(clusters["X"]),
+                        "y": list(clusters["Y"]),
+                        "z": list(clusters["Z"]),
+                    },
```

Comment:
It would be awesome if you could retain z-statistics and/or contrast values in the coordinate version of the dataset as well.

Reply:
done! (at least a first pass is done)